src/lib/ImitatorUtilities.ml

```diff
@@ -490,3 +489,31 @@ let terminate_program () =
 	flush stderr;
 	flush stdout;
 	exit(0)
+
+(* Misc *)
+
+(* Convert a syntatic array access (`x[i]`) into the identifier following our convention (`x___i`) *)
+let gen_access_id arr_name index = arr_name ^ "___" ^ (Int.to_string index)
```

Comment:
That's a good point. I did some tests in some parts of the code, but it would be better to move then to this function. Although we can't check if the index is greater than the length of the array, since this function is used once in the parser, before we know the length of the array. However, this usage is removed in the next PRs, so I think we can add this improvement on them?

Reply:
If it's done later, so it's OK ;)

Reply:
OK! I added a TODO here. This will be added in the `forall` PR.